### PR TITLE
Improve step slider precision

### PR DESCRIPTION
**Pull request by ricklove**

### CODE REVIEW

File: src/controls/widgets/number/InputNumberUI.tsx [hunk 1]
```diff
@@ -67,24 +70,30 @@ export const InputNumberUI = observer(function InputNumberUI_(p: {
...
         setInputValue(num.toString())
     }
 
     const mouseMoveListener = (e: MouseEvent) => {
+        // reset origin if change shift or control key while drag (to let already applied changes remain)
+        if (dragged && (lastShiftState !== e.shiftKey || lastControlState !== e.ctrlKey)) {
+            startValue = lastValue
+            cumulativeOffset = 0
+        }
+
```
Comment (ricklove) [inline src/controls/widgets/number/InputNumberUI.tsx:96]: This lets the user hit ctrl/shift while dragging without it jumping

File: src/controls/widgets/number/InputNumberUI.tsx [hunk 2]
```diff
@@ -276,7 +288,7 @@ export const InputNumberUI = observer(function InputNumberUI_(p: {
                                         return
                                     }
 
-                                    syncValues(ev.currentTarget.value, 0.001)
+                                    syncValues(ev.currentTarget.value, 0.000001)
```
Comment (ricklove) [inline src/controls/widgets/number/InputNumberUI.tsx:291]: Let the user type with highest precision

File: src/controls/widgets/number/InputNumberUI.tsx [hunk 3]
```diff
@@ -34,7 +37,7 @@ export const InputNumberUI = observer(function InputNumberUI_(p: {
     const val = p.value ?? clamp(1, p.min ?? -Infinity, p.max ?? Infinity)
     const mode = p.mode
     const step = p.step ?? (mode === 'int' ? 1 : 0.1)
-    const decimalPoints = Math.ceil(-Math.log10(step))
+    const rounding = Math.ceil(-Math.log10(step * 0.01))
```
Comment (ricklove) [inline src/controls/widgets/number/InputNumberUI.tsx:40]: Default rounding should be step/100

**Comment by birdddev** [comment]
LGTM uvu

**Pull request merged**